controllers/globals.go

```diff
@@ -0,0 +1,5 @@
+package controllers
+
+// Currently we only have 1 written and 1 reader so there is no reason to
```

Comment:
If both the reader and the writer access the map concurrently (which sounds possible?) then the code will panic. I would wrap this with a mutex now.

Reply:
@qbarrand why would code panic? at the worst case, etiher the reader won't find the key ( it's ok, will find it in the next iteration), or the read value will be nil value, which can easily be checked.

Reply:
@yevgeny-shnaidman https://go.dev/play/p/f9Klsm03D8C

Reply:
i see, Golang is not C , unfortunately. Ok, let's add ReadWriter lock

Reply:
I was not aware of that either. Will add a lock then.

Reply:
Done. Added a simple lock for simplicity. We can use 2 locks (one for each map) but it seemed redundant and make the code more complex and since node/imagestream events are rare then I don't see the justification for it.

Reply:
How is it redundant? Even though we do not expect a high rate of updates, blocking access to a data structure we're not touching doesn't sound right.

Reply:
Added a second mutex.